### PR TITLE
wait a little bit for data before going into curl pause

### PR DESCRIPTION
**Pull request by hassanctech**
*Issue #, if available:* https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/issues/490

*Description of changes:*

In the post read callback if I don't have data right away, I sleep and try again for 6 iterations with sleep [20 40 80 160 320] ms. My initial tests show this yields significant savings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hassanctech** [comment]
@rejeangroleau can you please try this and let me know if this resolves your issue?

File: src/source/Response.c [hunk 1]
```diff
@@ -635,18 +636,25 @@ SIZE_T postReadCallback(PCHAR pBuffer, SIZE_T size, SIZE_T numItems, PVOID custo
...
+        bytesWritten = (SIZE_T) retrievedSize;
 
-    bytesWritten = (SIZE_T) retrievedSize;
+        DLOGV("Get Stream data returned: buffer size: %u written bytes: %u for upload handle: %" PRIu64 " current stream handle: %" PRIu64,
+              bufferSize, bytesWritten, uploadHandle, pCurlResponse->pCurlRequest->streamHandle);
 
-    DLOGV("Get Stream data returned: buffer size: %u written bytes: %u for upload handle: %" PRIu64 " current stream handle: %" PRIu64, bufferSize,
-          bytesWritten, uploadHandle, pCurlResponse->pCurlRequest->streamHandle);
+        iter++;
+    } while(iter < 5 && bytesWritten == 0 && (retStatus == STATUS_SUCCESS || retStatus == STATUS_NO_MORE_DATA_AVAILABLE));
```
Comment (MushMal) [inline src/source/Response.c:657]: Define 5 in the response header file instead of hard coding

File: src/source/Response.c [hunk 2]
```diff
@@ -635,18 +636,25 @@ SIZE_T postReadCallback(PCHAR pBuffer, SIZE_T size, SIZE_T numItems, PVOID custo
         CHK(FALSE, retStatus);
     }
 
-    retStatus =
-        getKinesisVideoStreamData(pCurlResponse->pCurlRequest->streamHandle, uploadHandle, (PBYTE) pBuffer, (UINT32) bufferSize, &retrievedSize);
-
-    if (pCurlApiCallbacks->curlReadCallbackHookFn != NULL) {
+    do {
+        if(iter > 0) {
```
Comment (MushMal) [inline src/source/Response.c:640]: Space after if

**Review by MushMal — APPROVED** [review]
Ignorable minor comments

**Comment by rejeangroleau** [comment]
> @rejeangroleau @jlong29 can you please try this and let me know if this resolves your issue?
@hassanctech This works fine for our target platform, looks good to me!

**Review by disa6302 — APPROVED** [review]

**Pull request merged**